### PR TITLE
HDDS-1728. Add metrics for leader's latency in ContainerStateMachine. Contributed by Mukul Kumar Singh.

### PR DESCRIPTION
**Pull request by mukul1987**
This jira adds metrics around leaders latency for pipeline operations.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 33 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for branch |
| +1 | mvninstall | 500 | trunk passed |
| +1 | compile | 255 | trunk passed |
| +1 | checkstyle | 64 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 839 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 161 | trunk passed |
| 0 | spotbugs | 315 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 504 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 36 | Maven dependency ordering for patch |
| +1 | mvninstall | 460 | the patch passed |
| +1 | compile | 245 | the patch passed |
| +1 | javac | 245 | the patch passed |
| +1 | checkstyle | 65 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 621 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 148 | the patch passed |
| +1 | findbugs | 518 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 233 | hadoop-hdds in the patch passed. |
| -1 | unit | 1313 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 6233 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.web.client.TestBuckets |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1022/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1022 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux eacabed6637d 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / be80334 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1022/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1022/1/testReport/ |
| Max. process+thread count | 5317 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service hadoop-ozone/integration-test U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1022/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java [hunk 1]
```diff
@@ -43,14 +46,28 @@
...
   private @Metric MutableCounterLong numApplyTransactionFails;
   private @Metric MutableCounterLong numReadStateMachineFails;
   private @Metric MutableCounterLong numReadStateMachineMissCount;
+  private @Metric MutableCounterLong numStartTransactionVerifyFailures;
+  private @Metric MutableCounterLong numContainerNotOpenVerifyFailures;
 
   public CSMMetrics() {
+    int numEnumEntries = ContainerProtos.Type.values().length;
+    this.opsLatency = new MutableRate[numEnumEntries];
+    this.registry = new MetricsRegistry("CSMMetrics");
```
Comment (lokeshj1703) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java]: We can use CSMMetrics.class.getName() here.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 8 | https://github.com/apache/hadoop/pull/1022 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1022 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1022/2/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by lokeshj1703** [comment]
@mukul1987 Thanks for updating the PR. The changes look good to me. +1.

**Pull request merged**